### PR TITLE
Allow to exclude certain methods from Metrics/BlockLength

### PR DESCRIPTION
**Pull request by NobodysNightmare**
This PR introduces the easiest of my proposals made in issue https://github.com/bbatsov/rubocop/issues/3772

The idea is, that we do not need to exclude whole files from the cop `Metrics/BlockLength`, but instead can keep it activated for all files and instead allow some blocks to be larger.

I also changed the `defaults.yml` accordingly, however I assume that the exact method list there is highly opinionated. I tried to be conservative in only adding methods that are very common (RSpec, Rake, Rails) and that can legitimately receive a large block.

This list can surely be extended with more frameworks, however I am not sure if this is a good fit for the rubocop defaults.

### CODE REVIEW

**Comment by NobodysNightmare** [comment]
Organisational question (in case you agree with the change in general):

My change makes the `Metrics/BlockLength` cop fail on some occasions across the codebase. In my opinion correctly.

Should I fix those offenses right away? Or is it valid to simply add a new exclusion to the `rubocop_todo.yml`?

In any case I would wait for some first feedback before starting to fix the remaining code.

File: config/default.yml [hunk 1]
```diff
@@ -1172,10 +1172,13 @@ Metrics/MethodLength:
 Metrics/BlockLength:
   CountComments: false  # count full line comments?
   Max: 25
-  Exclude:
-    - 'Rakefile'
-    - '**/*.rake'
-    - 'spec/**/*.rb'
+  ExcludedMethods:
```
Comment (bbatsov) [inline config/default.yml]: I wouldn't add library specific stuff to the default config. This should be in the RuboCop-specific config.

**Comment by bbatsov** [comment]
> I also changed the defaults.yml accordingly, however I assume that the exact method list there is highly opinionated. I tried to be conservative in only adding methods that are very common (RSpec, Rake, Rails) and that can legitimately receive a large block.

As I've said in an inline comment - the default config should have nothing that's 3rd party-related. Everyone can adjust their config accordingly.

> Should I fix those offenses right away? Or is it valid to simply add a new exclusion to the rubocop_todo.yml? For now I did the latter.

If fixing the offenses in easy - it's usually to fix them alongside tweaking the cop.

File: config/default.yml (hunk 1 above)
Comment (NobodysNightmare) [inline config/default.yml]: Sure I can do that,

but I want to point out that excluding files ending in `.rake` or living in the folder `spec` also carries some library specfic assumptions.

Should I revert the config to its initial state (i.e. keep the file exclusions) or leave out both, the file exclusions and the excluded methods?

Comment (bbatsov) [inline config/default.yml]: Guess we should remove the exclusions completely and just modify RuboCop's own config.

Comment (NobodysNightmare) [inline config/default.yml]: Okay, that's what I did now:

* no exclusions (of any kind) in the default config
* rubocop uses the old (file based) exclusion for its own validation

**Pull request merged**